### PR TITLE
feat: add custom ArrayField component to form renderer

### PR DESCRIPTION
**Pull request by Jshhhh**
*Issue #, if available:*

*Description of changes:*
- Create ArrayField component for form fields with array type
- Adds validation for array type fields in onSubmit

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by hein-j — APPROVED** [review]

File: packages/test-generator/lib/generators/TestGenerator.ts [hunk 1]
```diff
@@ -119,8 +120,16 @@ export abstract class TestGenerator {
       try {
         if (this.params.writeToDisk) {
           const res = this.writeFormToDisk(schema as StudioForm);
-          if (res.formMetadata?.fieldConfigs && Object.keys(res.formMetadata.fieldConfigs).length) {
-            utilsFunctions.add('validation');
+          if (res.formMetadata?.fieldConfigs) {
+            const fieldConfigs = Object.values(res.formMetadata.fieldConfigs);
+            if (fieldConfigs.length) {
+              utilsFunctions.add('validation');
+              fieldConfigs.forEach(({ isArray }) => {
```
Comment (SwaySway) [inline packages/test-generator/lib/generators/TestGenerator.ts:127]: nit: might be faster to use a some

File: packages/codegen-ui-react/lib/react-utils-studio-template-renderer.ts [hunk 2]
```diff
@@ -60,24 +61,46 @@ export class ReactUtilsStudioTemplateRenderer extends StudioTemplateRenderer<
...
 
     this.utils.forEach((util) => {
       if (util === 'validation') {
         utilsStatements.push(...generateValidationFunction());
       } else if (util === 'formatter') {
         utilsStatements.push(...generateFormatUtil());
+      } else if (util === 'arrayField') {
+        skipReactImport = false;
+        this.fileName = `utils.${scriptKindToFileExtension(this.renderConfig.script)}`;
+        this.importCollection.addMappedImport(ImportValue.UI_REACT_STYLES);
```
Comment (SwaySway) [inline packages/codegen-ui-react/lib/react-utils-studio-template-renderer.ts:74]: the ui primitives don't import the styles I think thats expected from the customer if they want to use the provided amplify ui css styles
https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/src/primitives/TextField/TextField.tsx

Comment (Jshhhh) [reply packages/codegen-ui-react/lib/react-utils-studio-template-renderer.ts:74]: Yup realized that when rendering a test component. Closing this PR for rework

File: packages/test-generator/lib/generators/TestGenerator.ts (hunk 1 above)
Comment (Jshhhh) [reply packages/test-generator/lib/generators/TestGenerator.ts:127]: Will be removed in new PR

**Pull request closed without merging**